resources/home/dnanexus/run_workflows/tests/test_manage_dict.py

```
@@ -1481,3 +1422,152 @@ def test_missing_metrics_output_raises_assertion_error(self):
+        config = {"hold": True, "foo": {"hold": 0}}
+        assert is_held(config)
+
+    def test_tso500_config_with_hold(self):
+        """
+        Test when hold key is present in typical eggd_tso500
+        config structure
+        """
+        test_tso500_config_path = (
+            "resources/home/dnanexus/run_workflows/tests/data/build_job_inputs/tso500_config.json"
+        )
+        config = json.load(open(test_tso500_config_path))
```

Ruff (0.12.2)

Use a context manager for opening files

(SIM115)